### PR TITLE
Create regular participles in Dutch

### LINKED ISSUE #389: Create regular participle forms
PRs:
#397
Yoast/YoastSEO.js-premium-configuration/pull/73 (premium configuration)

We should create participle forms of regular Dutch verbs, i.e. those that are not on verb exception lists. This also includes participles with separable and inseparable prefixes, e.g. _omgedraaid_ or _behandeld_.

### PR DESCRIPTION
**Pull request by agnieszkaszuba**
## Summary

This PR can be summarized in the following changelog entry:

Creates participles of regular Dutch verbs. 

## Test instructions

This PR can be tested by following these steps:

Add more examples to the specs. The following should be covered:

* Participles that take the -_t_ or -_d_ suffix (or no suffix if the stem already ends in -_t_ or -_d_) and that take the _ge_ prefix, e.g. _toon_ -> _getoond_, with an optional separable prefix (e.g. _om_) that comes before the _ge_ prefix, e.g. _omdraai_ -> _omgedraaid_
* Participles of compound verbs with an inseparable prefix (e.g., _be_) that take the _-d_ suffix, e.g. _behandel_ -> _behandeld_. Participles of inseparable compound verbs with the _-t_ suffix or with no suffix are the same as other verb forms (1st or 3d person singular) so they are already covered.
* Exception verbs that have a separable prefix which in that specific verb is inseparable (e.g., _omringd_) are also covered. The exception list can be found in the morphology data file under `inseparableVerbs`.

The following are NOT covered:
- participles of verbs that are on verb exception lists, e.g. participles of strong verbs (these will be covered later)
- participles with uncommon separable prefixes, e.g. _zwart_ as in _zwartgemaakt_. The list of separable prefixes that are currently taken into account can be found in the morphology data under `compoundVerbsPrefixes` -> `separable`.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #389

### CODE REVIEW

File: packages/yoastseo/src/morphology/dutch/generateParticipleForm.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+ *  as inseparable. The exception list includes only those stems that need to take the -d suffix. Those that need
+ *  the -t suffix or no suffix are the same as another form (1st or 3rd person singular) created elsewhere.
+ *
+ * @param {string} stemmedWord The stem to check.
+ * @param {string[]} inseparableExceptions	The list of exception stems.
+ * @returns {string|null} The created participle form or null if the stem was not found on the exception list.
+ */
+const generateInseparableExceptionParticipleForm = function( stemmedWord, inseparableExceptions ) {
+	if ( inseparableExceptions.includes( stemmedWord ) ) {
+		return addParticipleAffixes( stemmedWord, "d" );
```
Comment (hansjovis) [inline packages/yoastseo/src/morphology/dutch/generateParticipleForm.js]: I would simplify it to this:
```suggestion
		return stemmedWord + "d";
```
This communicates directly to the reader of the code that we add the `d` suffix to the end of the stemmed word. Wrapping it into a separate function adds a layer of indirection, in my opinion.

File: packages/yoastseo/src/morphology/dutch/generateParticipleForm.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+ * @returns {string|null} The created participle form or null if the stem doesn't start with a prefix.
+ */
+const generateParticipleFormWithSeparablePrefix = function( morphologyDataVerbs, stemmedWord, suffix, prefixes ) {
+	for ( const currentPrefix of prefixes ) {
+		if ( stemmedWord.startsWith( currentPrefix ) ) {
+			const stemmedWordWithoutPrefix = stemmedWord.slice( currentPrefix.length, stemmedWord.length );
+
+			const prefix = morphologyDataVerbs.participleAffixes.prefix;
+
+			return addParticipleAffixes( stemmedWordWithoutPrefix, suffix, prefix, currentPrefix );
```
Comment (hansjovis) [inline packages/yoastseo/src/morphology/dutch/generateParticipleForm.js]: The same here:
```suggestion
			return currentPrefix + prefix + stemmedWordWithoutPrefix + suffix;
```

File: packages/yoastseo/src/morphology/dutch/generateParticipleForm.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+		stemmedWord,
+		morphologyDataVerbs.compoundVerbsPrefixes.inseparable
+	);
+
+	/** If an inseparable prefix is found, the participle form should only be created if the stem takes the -d suffix
+	 * in the participle. If it takes the -t suffix or on suffix, then it is the same as the third person singular present
+	 * form so there is no need to create the participle form. */
+	if ( inseparablePrefix ) {
+		if ( suffix === "d" ) {
+			return addParticipleAffixes( stemmedWord, suffix );
```
Comment (hansjovis) [inline packages/yoastseo/src/morphology/dutch/generateParticipleForm.js]:
```suggestion
			return stemmedWord + suffix;
```

**Comment by agnieszkaszuba** [comment]
@hansjovis 
I have deleted the addParticipleAffixes function for Dutch entirely since it doesn't seem necessary according to your comments :)
Also, the addParticipleAffixes function that was put in the helpers is now moved back to its original file, the one for creating German participles. I had moved it to helpers because I thought it could be used for Dutch, but this turned out to not be the case.

**Comment by hansjovis** [comment]
Acceptance test: ✅ .

I added some specs to test the code and committed them as well. Will wait until the Travis build passes before merging the branch.

**Pull request merged**